From 0dfb785d31cad8d0b48311dfa0a15916ab54d6f8 Mon Sep 17 00:00:00 2001
From: Yifan Hong <elsk@google.com>
Date: Mon, 9 Oct 2017 15:21:15 -0700
Subject: [PATCH] Fix BatteryManager.getIntProperty returns 0 as invalid value.

Long.MIN_VALUE cast to Integer.MIN_VALUE always returns 0.

diff --git a/core/java/android/os/BatteryManager.java b/core/java/android/os/BatteryManager.java
index f715f50..9eccb37 100644
--- a/core/java/android/os/BatteryManager.java
+++ b/core/java/android/os/BatteryManager.java
@@ -287,7 +287,8 @@
      * @return the property value, or Integer.MIN_VALUE if not supported.
      */
     public int getIntProperty(int id) {
-        return (int)queryProperty(id);
+        long val = queryProperty(id);
+        return val == Long.MIN_VALUE ? Integer.MIN_VALUE : (int)val;
     }
 
     /**
